### DISCUSSION TITLE
go iterator ergonomics improvement

### DISCUSSION THREAD

**Discussion by wkalt** [foxglove/mcap#1077] [General]
It would be nice to introduce some kind of `More()` style convenience like in the rosbag reader example here: https://github.com/foxglove/go-rosbag#reading-a-bag-file

I think this can probably be done backward compatibly with the existing calling style, which is to break out of the loop on EOF. If using More-style, the EOF-producing call would not be made.

Let me know if it seems desirable and I may raise a patch.

**Comment by defunctzombie** [comment]
Is this _More_ convention something other file readers in golang do? My initial impression (coming from what we do in typescript) is that `.Next()` returns stuff to read or an indication that there's no more stuff to read so we typically iterate on _next_ and avoid the sort of "double call" to query if there is more and then read more.

I do think have a _More_ (or I've seen it called _Peek_ in other apis) is useful on its own even when its still nice to use _next_ to do the iteration.
